k8s/overlays/test/fixture_mount.yaml

```diff
@@ -0,0 +1,55 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- path: /kind
+  op: replace
+  value: StatefulSet
+
+- path: /spec/template/spec/volumes/1
+  op: add
+  value:
+    name: test-fixture
+    configMap:
+      name: collector-config
+      items:
+        - key: fixture.json
+          path: fixture.json
+
+- path: /spec/template/spec/volumes/2
+  op: add
+  value:
+    name: output
+    emptyDir: {}
+
+- path: /spec/template/spec/containers/1
```

Comment:
I'm not sure I completely understand why we needed this second container. 

From the `Makefile` it looks like we only use this to pull the output out, but can we not do that from the otel container directly?

(fyi this maybe some idiomatic k8s thing that I have no idea about so sorry if its something obvious :))

Reply:
Yeah, this is kind of a weird k8s thing. The `kubectl cp` command that we use to copy the file out of the container uses `tar` to extract the files from the container. The `opentelemetry-collector` image doesn't have this, so we add the sidecar in here to give us side-door access to the container filesystem.

Trying to copy the file right from the opentelemetry-collector container gives this error:
```
$ kubectl cp -c opentelemetry-collector opentelemetry/opentelemetry-collector-0:/output/output.json test.json
error: Internal error occurred: error executing command in container: failed to exec in container: failed to start exec "90d7b009a6cd8d7447378bd44b63a648cc77d8fd2b92091872202331aaedc1b1": OCI runtime exec failed: exec failed: unable to start container process: exec: "tar": executable file not found in $PATH: unknown
```